### ISSUE TITLE
[Feature Request] Add an option to set your own descriptor resolver

### ISSUE THREAD

**Issue by rauanmayemir** [bufbuild/protovalidate-go#162]
**Feature description:** 

Similar to `WithExtensionTypeResolver`, it would be helpful to allow adding your own `WithDescriptorResolver` option.

**Problem it solves or use case:** 

It will allow using dynamic schema registries outside of global `protoregistry.Files`

**Proposed implementation or solution:** 

Something like:
```
func WithDescriptors(resolver ...protodesc.Resolver) ValidatorOption {
  return func(cfg *config) {
    cfg.descResolver = resolver
  }
}
```

**Comment by jhump** [comment]
@rauanmayemir, the only way a descriptor resolver might be used is for CEL expressions that operate on `google.protobuf.Any` messages. Is that the issue you are encountering? If not, please describe the issue you are encountering with dynamic schemas, in case you're running into something else (like a bug).

**Comment by rauanmayemir** [comment]
@jhump No, that wasn't my issue. Apparently, I didn't even have to add my regular message descriptors because the input message already has a ProtoReflect() with all the information, is that correct?

**Comment by jhump** [comment]
That's correct. A descriptor resolver is generally unnecessary since the message being validated knows its own descriptor (and all fields therein know their message types, too).

The only time you would run into trouble is when using `google.protobuf.Any` messages, where the contained message payload is a dynamic message (i.e. _not_ known to the program via `protoregistry.GlobalFiles`). In this case, there are some operations in CEL expressions that will try to unpack the `Any` message, which would fail in this case. This is the only case where you would benefit from an option like this, to provide additional dynamic message types that can be unpacked to the CEL interpreter.

**Comment by jchadwick-buf** [comment]
I think there is no issue here as it should indeed work just fine to use different descriptor registries with protovalidate. Please feel free to open another issue or comment here if there are any problems still.

**Comment by rauanmayemir** [comment]
There is no issue, but the feature request was still valid for when you want to add extra descriptors for Any messages.
